### PR TITLE
Add log explorer

### PR DESCRIPTION
**Pull request by jwoudenberg**
This adds a TUI for exploring log spans produced by `nri-prelude` code. It's out-of-the-box able to read log data from `nri-prelude` tests, but can be used for logs coming from applications running in development too if the application has been instrumented appropiately.

To test it out:

```
# run tests to produce some data
cabal test
# start log explorer
cabal run log-explorer
```

You can also run these commands in reverse order. The log explorer will update with new logs when they happen.

It's still pretty basic, but I think useful enough to be released as a v1!

### CODE REVIEW

**Comment by michaelglass** [comment]
neat

File: nri-log-explorer/src/Main.hs [hunk 1]
```diff
@@ -0,0 +1,624 @@
...
+        case fn (toPage model) of
+          NoLogsFound -> model {selectedRootSpan = Nothing, loglines = Nothing}
+          SpanList _ zipper -> model {selectedRootSpan = Nothing, loglines = Just zipper}
+          SpanDetails _ _ zipper -> model {selectedRootSpan = Just zipper}
+   in newModel {userDidSomething = True}
+
+viewKey :: Page -> Brick.Widget Name
+viewKey page =
+  let exit = "q: exit"
+      updown = "↑↓: select"
```
Comment (stoeffel) [inline nri-log-explorer/src/Main.hs:226]: `jk` would be nice :-)

Comment (jwoudenberg-nri) [reply nri-log-explorer/src/Main.hs:226]: `jk` is supported! I just left it out of the help line to keep it short!

**Review by stoeffel — APPROVED** [review]
This is great!

**Pull request merged**